### ISSUE TITLE
This datetime string won't parse, and I have no idea why

### ISSUE THREAD

**Issue by zacarybrown** [crsmithdev/arrow#354]
```
>>> arrow.get('2016-03-01T07:45:12.6842351+00:00')
Traceback (most recent call last):
.
.
.
  File "/python_env/adt/lib/python3.5/site-packages/arrow/arrow.py", line 290, in __str__
    return self._datetime.isoformat()
ValueError: offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24).
```

And yet this works fine (I just removed the right-most digit from the sub-seconds)

```
>>> arrow.get('2016-03-01T07:45:12.684235+00:00')
<Arrow [2016-03-01T07:45:12.684235+00:00]>
```

And so does this (the original string with one digit added to the sub-seconds):

```
>>> arrow.get('2016-03-01T07:45:12.68423510+00:00')
<Arrow [2016-03-01T07:45:12.684235+10:00]>
```

What's going on here?

**Comment by pypingou** [comment]
FTR, I'm getting a different output:
~~~~
>>> import arrow
>>> arrow.get('2016-03-01T07:45:12.6842351+00:00')
<Arrow [2016-03-01T07:45:12.000684+23:51]>
>>> arrow.get('2016-03-01T07:45:12.684235+00:00')
<Arrow [2016-03-01T07:45:12.684235+00:00]>
>>> arrow.get('2016-03-01T07:45:12.68423510+00:00')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    arrow.get('2016-03-01T07:45:12.68423510+00:00')
  File "/usr/lib/python2.7/site-packages/arrow/arrow.py", line 286, in __repr__
    return '<{0} [{1}]>'.format(self.__class__.__name__, self.__str__())
  File "/usr/lib/python2.7/site-packages/arrow/arrow.py", line 289, in __str__
    return self._datetime.isoformat()
ValueError: tzinfo.utcoffset() returned 1440; must be in -1439 .. 1439
~~~~

**Comment by zacarybrown** [comment]
Stepping thru the code, it appears that arrow.get() doesn't currently support strings that have more than 6 digits of sub-seconds.  However, it also doesn't raise any explicit errors when presented with these strings.  Instead, it treats the subsequent digits as the timezone offset.  

To see this behavior, try running the last example that I presented in my first post:

>>> arrow.get('2016-03-01T07:45:12.68423510+00:00')
<Arrow [2016-03-01T07:45:12.684235+10:00]>

Note that the resulting arrow object has an offset of 10, which is actually the last two digits of the sub-seconds.

Sometimes this results in errors, depending on what those remaining sub-second digits contain.  In any case, it's not likely what the caller wants, since those digits were not intended to be interpreted as a timezone offset.

**Comment by bukzor** [comment]
I believe this is fixed in 0.9.0.
Please have a try.

**Comment by andrewelkins** [comment]
Closing. Please reopen if it is still an issue
